### DISCUSSION TITLE
No such interface “org.freedesktop.portal.RemoteDesktop” on object at path /org/freedesktop/portal/desktop

### DISCUSSION THREAD

**Discussion by krims0n32** [rustdesk/rustdesk#8935] [Q&A] [answered]
### Bug Description

When connecting to a host in the same network using IP I get this error message after entering the password or pressing Accept on the other side.

### How to Reproduce

Allow Direct IP connection on remote host.
Connect using Direct IP to remote host. Enter password or press accept on other side.

### Expected Behavior

Expect the session to be established without error.

### Operating system(s) on local side and remote side

Fedora 40 (arm64) -> Arch Linux

### RustDesk Version(s) on local side and remote side

1.2.7 -> 1.2.7

### Screenshots

[Screencast_20240802_171711.webm](https://github.com/user-attachments/assets/fa2d2e48-c846-45c7-a5aa-70a26d58a41e)

### Additional Context

Installed using RPM, flatpak or appimage. Error is the same.

**Comment by AgostinoA** [comment]
I confirm it happens to me too with xdg-desktop-portal-wlr + wayland + wlroots.
I recommend using org.freedesktop.impl.portal.ScreenCast with Wayland
For Teams, Telegram and other apps, Screencasting works great

**Reply by rustdesk** [reply]
org.freedesktop.impl.portal.ScreenCast is good for taking one picture (this api save one file on hard drive every time, you can imagine how slow it is) , but not for continuously streaming.

**Reply by AgostinoA** [reply]
I fully understand your point, but integrating a dual solution when no other option is available, would resolve compatibility issues from different DDEs and xdg-desktop-portals implemetations, 5 minutes of clock time is needed to implement this change.

**Comment by vincentaxhe** [comment]
hyprland, rpi kde, iphone face the same problem

**Comment by ron-1986** [comment]
I have the same problem.
Wanted to connect from 

Arch Linux / XFCE desktop Rustdesk 1.37(Arch)
to
Rustdesk Appimage 1.37 on a Gnome Desktop (OS sadly unknown)

Getting this error:
No such interface “org.freedesktop.portal.RemoteDesktop” on object at path /org/freedesktop/portal/desktop

Is there any way to workaround this and connect? I can connect fine to Windows Desktops and some other Linux desktops.

**Comment by Sepzilla** [comment]
I get the same error, from arm MacOS to Cosmic 24.04. direct connection enable. Have tried IP and IP+direct lan port.

One thing to add though, even trying for a file transfer connection also craps out and gives "No active console user logged on, please connect and logon first." I have terminal open on both ends, but I don't think that's relevant...

Build 1.3.8 on both machines, local relay on a synology through docker.

**Comment by AgostinoA** [comment, marked as answer]
[xdg-desktop-portal-wlr-0.7.0.r11.g1eaa02e-1-x86_64.pkg.tar.zst.zip](https://github.com/user-attachments/files/20525944/xdg-desktop-portal-wlr-0.7.0.r11.g1eaa02e-1-x86_64.pkg.tar.zst.zip)
(remove .zip)

[https://github.com/David96/xdg-desktop-portal-wlr](https://github.com/David96/xdg-desktop-portal-wlr)

I compiled a version that at least partially resolves RemoteDesktop sharing. In my case, the virtual mouse doesn’t work — I need to apply some fixes with wlroots

https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4980/diffs?file=43f78b67a7960d2c6c8503cd07ae726214fa9f38#diff-content-43f78b67a7960d2c6c8503cd07ae726214fa9f38

**Reply by onyxiarivera** [reply]
This fixed my issues being unable to connect on Hyprland, mouse works as well.
